extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -146,14 +147,22 @@ private List<InputRow> parseMetric(Metric metric, Map<String, Object> resourceAt
         inputRows = new ArrayList<>(metric.getSum().getDataPointsCount());
         metric.getSum()
             .getDataPointsList()
-            .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
+            .forEach(dataPoint -> {
+              if (hasRecordedValue(dataPoint)) {
+                inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName));
+              }
+            });
```

Comment:
did you do this for performance reasons vs.

```suggestion
            .stream().filter(hasRecordedValue)
            .forEach(dataPoint -> {
                inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName));
            });
```

Reply:
Yes, I thought @xvrl had made a comment about that somewhere that I wanted to link to but couldn't find it

Reply:
https://github.com/confluentinc/druid/pull/157#discussion_r1336426519